modules/afsocket/afsocket-dest.c

```diff
@@ -335,6 +337,11 @@ afsocket_dd_construct_writer_method(AFSocketDestDriver *self)
 static gboolean
 afsocket_dd_setup_writer(AFSocketDestDriver *self)
 {
+  afsocket_dd_restore_writer(self);
+
+  if (self->writer && log_pipe_is_initialized((LogPipe *)self->writer))
```

Comment:
self->writer can't be initialized at this phase, so this check is unnecessary, I think